### PR TITLE
Adds an SM shard to the derelict.

### LINKED ISSUE #2287: SM Shard for Derelict
Please add it.. for the love of the autism MoMMI's

### LINKED ISSUE #3420: Adding supermatter shard to derelict - discussion
A supermatter shard on derelict would allow mommies to learn how to setup supermatter without destroying up the main station, it would also provide more fun for players removed from the round.

Currently shards can be obtained from cargo for 200 a piece, for comparison a cheap lighter from a vendor costs 60. They can be spammed to hell and back, there's one available roundstart in storage.

This is to illustrate that it would in fact not increase the likelyhood of shard abuse for antagonistic activities and as such would not alter balance to any notable degree. The only thing it would do is provide more fun for derelict mommies.

I will most likely add this myself unless someone has some reasoning as to why it is a bad idea.

### PR DESCRIPTION
**Pull request by PJB3005**
resolves #3420

resolves #2287

Also I have no idea why the first 2 changes lines on box are just ';' removals, DM must've done that automatically.

### CODE REVIEW

**Comment by ComicIronic** [comment]
I stand by no.

**Comment by PJB3005** [comment]
I can make the crate they're in locked in such a way only MoMMIs can open them.

Also the SME rad wave when it explodes now only affects the Z level it's on (as placed in the changelog) so derelict MoMMIs fucking up won't kill the round.

File: code/modules/supermatter/supermatter.dm [hunk 1]
```diff
@@ -184,6 +184,8 @@
 		if(damage > explosion_point)
 			for(var/mob/living/mob in living_mob_list)
 				if(istype(mob, /mob/living/carbon/human))
+					if(mob.z != src.z)//only make it effect mobs on the current Z level.
```
Comment (ComicIronic) [inline code/modules/supermatter/supermatter.dm]: This shouldn't inside the human if, because the rad damage will still be applied.

Comment (PJB3005) [inline code/modules/supermatter/supermatter.dm]: oops

**Comment by Probe1** [comment]
No.

**Comment by AutismSpider** [comment]
don't just say no give a fucking explain as to why you don't want it on the derelict

**Comment by AutismSpider** [comment]
If its about the RAD crap when it blows

>PJB3005
>Also the SME rad wave when it explodes now only affects the Z level it's on (as placed in the changelog) so derelict MoMMIs fucking up won't kill the round.

**Comment by Probe1** [comment]
I've had it out in irc and the discussion issues before but I'll repeat it here.  Two points of contention and one alternative I'm fine with.

1.  Don't give dangerous shit to the derelict what the fuck.  Giving a SME or Singlo to the derelict is begging for inexperienced players to fuck up the main station.  The derelict is one part tutorial island and one part I died and am bored.  Giving a shard that can explode and give everyone a very loud alert, radiation and hallucinations to players who are new or bored is an idea I disagree with.  

If it delaminates everyone will get the warning and the effects.  So you'll have engineers on a goose chase looking for this shard that doesn't exist on their z level.  They won't know that.  The crew that feels the SM shard exploding is a negative part of their day will panic.  Meanwhile a dead player that spawned on the derelicts having fun fucking with everyone else.

2.  It's a place for dead players.  While some allowances should certainly be made, they are dead.  There should not be virgins in heaven waiting for them.  That only devalues life by giving incentives to players out of the main round, and rewards players that have been knocked out.  Having ways to rejoin the round that are part of the main round are fine.  Having a derelict where you can practice building and learn atmospherics without hurting anyone is also fine.  But giving the dead their own place where they have a end game engine is excessive.  It's rewarding people who have died.  

Alternative:  Don't merge this and let the status quo be - players can ahelp and receive a shard on the derelict if they feel ready for the task.  It's worked fine in the past and will work fine in the future.  Do add a TEG engine to the derelict and let them fuck with that.  It is an engine that receives little love in the current meta and having it there for people to train on it will increase its use.  It also is not catastrophic if it fails.  I would prefer nothing is added but if I had to settle, I would accept a TEG being added instead.

**Comment by PJB3005** [comment]
number 1: addressed.
number 2: it's not, if it would be engineering would be overstaffed every round, there'd be millions of MoMMIs, and the derelict would have functional atmos every other shift.

Though a TeG is fine by me

**Comment by Probe1** [comment]
You're not addressing it by nerfing the supermatter shards already almost irrelevant ill effects from delamination.  That's even further from what I agree with.

**Comment by FudgePucker** [comment]
The problem with derelict MoMMIs is that for most projects they depend on
A) The on station MoMMIs sending them things.
B) The gods blessing them with what they need.
The problem with this is that the gods are busy, and having to summon a shard every round can get tedious.

The problem with having a shard on the derelict is that shitters and new players might effect the round by causing the shard to blow up and effect the other zlevels, or teleport the shard onto the bridge as it is blowing up.

I have a somewhat simple solution to this, although it might take more work than just putting a shard on the derelict.

A computer that is usable by MoMMIs only, that spawns objects based upon a white list, meaning that if a good engineer wants a SM shard to make a SME, they can. And really good engineers who have proven themselves as not shitty can spawn in 300 collectors and emitters with out causing god to just gib them for asking for so much shit.

Another problem with derelict MoMMIs is that some objects are really hard to teleport in, A TEG requires 3 crates, and the last time I tried getting them from the teleporter, they only sent one regulator, and then the other one that they ordered after that suffered teleport error and ended up in the middle of space. I wasn't able to get my TEG up and running by the time the round ended. With this set up I am asking for, all a MoMMI has to do is press three buttons to start on their TEG.

Maybe have avaliable at round start is a TEG, and then they can get whitelisted for things as they prove themselves to the admins as not shitty.

Uranium silver gold and plasma can be whitelisted if they want to make better stock parts or PACMANs. Plastic can be whitelisted if they want to experiment or build with plastic flaps. Seed crates can be whitelisted in if they want to experiment with botany.

This really can't be abused because it is whitelist only, so that people can't teleport in infinite plastic, and have to prove that they won't teleport plastic back to the station before getting whitelisted, and if they do, well you can just ban them from MoMMI or remove them from the whitelist.

**Comment by AutismSpider** [comment]
>good engineers

Damn nigger that's good joke

**Comment by Probe1** [comment]
It's not half as funny as the mommi feature creep.  We should cut the bullshit and just give mommis admin build mode.

**Comment by FudgePucker** [comment]
>le MoMMI's feature creep
Oh wow, that was the one downside to my entire idea.
Fuck rewarding good players with a gold star, that lets them experiment to get better.
People in the thread would bitch about it.

**Comment by Probe1** [comment]
relax fudgepucker it was a joke in turn about spiders engineer sleight.  it wasn't a serious comment about your idea.

i'm adamant that the supermatter shard and singularity should not be added to the derelict under any circumstances but I have stated this and explained why.  i've said my part and you are cool to say yours.

**Comment by AutismSpider** [comment]
Singularity should be no it can hop form z level to z level

any shard spawn on derelict should be Z level lock like the mommis so if they try space it or teleport it just comes back to them.

**Comment by FudgePucker** [comment]
Sorry for being hostile Probe, I am still in thread level of hostility.
The whole point of the idea is pretty much this.
>players can ahelp and receive a shard on the derelict if they feel ready for the task.

The idea behind my proposal is that admins still have to decide if they are ready for the task, only they also get the benefit of not having to do it every round the same player wants to make a SME on the derelict. If the admins know that a player knows what they are doing, from building SMEs on the station before, they can let the player have the privilege of summoning one. If a player is good enough for them to give them a shard if they pray for it every time, then usually they can handle getting one with out admin intervention. In fact one time I asked over AHelp for someone to check out my set up, and they just gave me a full crystal, not even half way done setting it up and they trusted me enough with a crystal. This proposal would be for trustworthy players who know what they are doing.

In any case, I would prefer having a TEG on the derelict instead of a shard as well. Asking god for a shard is easy enough, asking god to spawn three objects is a little much.

**Comment by clusterfack** [comment]
If a mommi wanted to there would be any number of ways he could teleport terrible things to the station that would ruin everything, the gas miners being the first and foremost example. And if a player who died is mad and wants revenge, he could just as easily join as a diona to fuck up a round as he could a derelict mommi, with considerably less skill required as well. In the end you wont see this abused the same reason you dont see 99% of extremely abuse-able things on the station getting abused, due to the threat of getting banned.

I consulted with PJB over concerns about adding a shard to the derelict, namely the radiation burst as well as the fact the shard monitor broadcasts over zlevel 1 only, and he has addressed both concerns. This isn't a willy nilly PR, and hell the derelict on defficiency already has had a shard since very close to its inception, it was the only change I can think of duny making to separate it from box's derelict.

At worst your 'someone teleports an sme back to the station' happens once - the player gets silicon permabanned and a week server ban, then nobody does it again.

**Comment by PJB3005** [comment]
^ My thoughts exactly.

**Comment by DeityLink** [comment]
>Probe1
>The derelict is one part tutorial island and one part I died and am bored.

0 part treasure island makes me sad :<

**Comment by ComicIronic** [comment]
>yarr, gents, we be going to get da boo-
>oh, there's nothing here worth taking
>never mind, let's go home

Treasure Island: Pirates on the Derelict

**Comment by 9600bauds** [comment]
>float through space and end up in Derelict
>hack exterior airlock to try to get inside
>derelict MoMMIs wired the SME to grid because it's the only admin-approved way they have of indirectly killing people
epic

**Comment by AutismSpider** [comment]
>Hacking the airlocks to autism land

Cum on guys why would go near the or try to open it.

**Comment by clusterfack** [comment]
As if people dont just break all the windows in and out of every room on the derelict, even when the mommis have already pressurized it.

**Comment by 9600bauds** [comment]
>why would go near the or try to open it.

Because there's useful stuff in there I want. Derelict is not exclusively MoMMI kingdom.

**Comment by AutismSpider** [comment]
Only thing that's useful on the derelict is the R&D lab and the wooden stick

**Comment by 9600bauds** [comment]
And the security cameras computer, and the emitters, and CE's jumpsuits, an EMP grenade, and the hacked autolathe, and yes, the wooden stick and the R&D lab that are pretty important too.
MoMMIs don't have an use for most of that stuff.

**Comment by ComicIronic** [comment]
You forgot the suspicious toolbox.
Gotta max those levels.

**Comment by FudgePucker** [comment]
This is going to be totally unrelated.
But I think the super cool mostly unused department based venders would be cool to use on mini low pop maps.
The engineering one that is in the derelict, the unused unique sprite less toxins one.
Just remove the chief engineers jumpsuits and you could possibly put it on the station right now.

**Comment by Probe1** [comment]
I have never understood why there are four CE jumpsuits on the derelict vending machine but when I brought it up pomf said whatever.  Someone should patch it out anyways ~~or do the job properly and make interesting, engaging, and difficult antag objectives instead of stealing laundry~~.

**Comment by FudgePucker** [comment]
I mean, having the cooler engivend would be awesome to have on the station.
Also the CE jumpsuits look cool as shit, but being an antag objective means no fun allowed. Also it needs more hard hat flavors, it has CE jumpsuits but no CE hardhats for MoMMIs to wear.
Maybe remove the CE jumpsuit and add different flavors of engineering jumpsuits.

**Comment by Probe1** [comment]
There are sprites for orange hardhats.  They used to be in the construction area of deff.  This is outside the discussion of this PR but it would be simple to add them into vends.

**Comment by clusterfack** [comment]
Clearly the hacked vendor should be dispensing different kinds of engineering hats instead of jumpsuits

**Comment by DeityLink** [comment]
@Probe1 Don't fucking remove those CE Jumpsuits from Derelict :< I might have lost because of some petty shuttle griff butt that gave me a reason to try and seek out the Derelict when I accidentally fed the CE's locker to the singulo yesterday.

**Comment by PJB3005** [comment]
YES WE AT THE DELTA 3 DIFFS NOW

**Comment by PJB3005** [comment]
well fuck

**Comment by DrCelt** [comment]
Firmly for. No disadvantages, many advantages. People have to learn to make a SME and doing so on the main station is dangerous; it's better to fuck up the derelict than the main station.

**Comment by PJB3005** [comment]
Oh for fucks sake.

I JUST downloaded the current bleeding-edge map.

then I manually edited the 2 lines.

AND IT'S STILL 13000 DIFFS.

**Comment by PJB3005** [comment]
OH WHAT THE FUCK.

I JUST DOWNLOADED THE MAP, BOX IS THE CURRENT BLEEDING EDGE VERSION

STILL 13 K DIFFS.

I GIVE UP

**Comment by Probe1** [comment]
@DrCelt don't tell anyone but bombs made in toxins are far, far more dangerous than the supermatter shard delaminating.  Full crystal is a different deal.

**Pull request closed without merging**